src/test/java/uk/gov/justice/digital/delius/controller/api/AppointmentControllerTest.java

```diff
@@ -0,0 +1,298 @@
+                .when()
+                .get("/api/offenders/crn/CRN1/appointments")
+                .then()
+                .statusCode(200)
+                .extract()
+                .body()
+                .as(Appointment[].class);
+
+        verify(appointmentService).appointmentsFor(eq(1L), appointmentFilterArgumentCaptor.capture());
+
+        assertThat(appointmentFilterArgumentCaptor.getValue().getAttended().orElse(null)).isEqualTo(Appointment.Attended.NOT_RECORDED);
+        assertThat(appointmentFilterArgumentCaptor.getValue().getFrom().orElse(null)).isEqualTo(LocalDate.of(2018, 11, 2));
+        assertThat(appointmentFilterArgumentCaptor.getValue().getTo().orElse(null)).isEqualTo(LocalDate.of(2018, 12, 25));
+
+
+    }
+
+    @Test
+    public void filterIsOptionalWhenSuppliedWithCrn() {
+        when(offenderService.offenderIdOfCrn("CRN1")).thenReturn(Optional.of(1L));
+        when(appointmentService.appointmentsFor(eq(1L), any(AppointmentFilter.class)))
+                .thenReturn(ImmutableList.of(aAppointment(2L, "Drugs Checkup"), aAppointment(1L, "Accommodation")));
+
+        given()
+                .when()
+                .get("/api/offenders/crn/CRN1/appointments")
+                .then()
+                .statusCode(200)
+                .extract()
+                .body()
+                .as(Appointment[].class);
+
+        verify(appointmentService).appointmentsFor(eq(1L), appointmentFilterArgumentCaptor.capture());
+
+        assertThat(appointmentFilterArgumentCaptor.getValue().getAttended().isPresent()).isFalse();
+        assertThat(appointmentFilterArgumentCaptor.getValue().getFrom().isPresent()).isFalse();
+        assertThat(appointmentFilterArgumentCaptor.getValue().getTo().isPresent()).isFalse();
+    }
+
+    @Test
+    public void filterIsPopulatedWhenSuppliedWithNoms() {
+        when(offenderService.offenderIdOfNomsNumber("NOMS1")).thenReturn(Optional.of(1L));
+        when(appointmentService.appointmentsFor(eq(1L), any(AppointmentFilter.class)))
+                .thenReturn(ImmutableList.of(aAppointment(2L, "Drugs Checkup"), aAppointment(1L, "Accommodation")));
+
+        given()
+                .params(ImmutableMap.of("from", "2018-11-02", "to", "2018-12-25", "attended", "NOT_RECORDED"))
+                .when()
+                .get("/api/offenders/nomsNumber/NOMS1/appointments")
+                .then()
+                .statusCode(200)
+                .extract()
+                .body()
+                .as(Appointment[].class);
+
+        verify(appointmentService).appointmentsFor(eq(1L), appointmentFilterArgumentCaptor.capture());
+
+        assertThat(appointmentFilterArgumentCaptor.getValue().getAttended().orElse(null)).isEqualTo(Appointment.Attended.NOT_RECORDED);
+        assertThat(appointmentFilterArgumentCaptor.getValue().getFrom().orElse(null)).isEqualTo(LocalDate.of(2018, 11, 2));
```

Comment:
we would have preferred a json test, but lets not change that now - out of scope and the risk is so low

Reply:
Yeah ideally.  I'm a little worried about scope creep it's true - don't want this task to carry on forever - but when I'm done let's take another look at other such improvements.